File: src/src/gestorAplicacion/libreria/EstadoEjemplar.java
package gestorAplicacion.libreria;

import gestorAplicacion.servicios.Prestamo;
import gestorAplicacion.servicios.Reserva;

import java.io.Serializable;

public class EstadoEjemplar implements Serializable {
    private boolean prestado;
    private boolean reservado;
    private boolean retraso;
    private Prestamo prestamo;
    private Reserva reserva;

    //Estado en el que se puede encontrar un ejemplar (libro o revista)
    public EstadoEjemplar(boolean prestado, boolean reservado, boolean retraso, Prestamo prestamo, Reserva reserva) {
        this.prestado = prestado;
        this.reservado = reservado;
        this.retraso = retraso;
        this.prestamo = prestamo;
        this.reserva = reserva;
    }

    public boolean isPrestado() {
        return prestado;
    }

    public boolean isReservado() {
        return reservado;
    }

    public boolean isRetraso() {
        return retraso;
    }

    public Prestamo getPrestamo() {
        return prestamo;
    }

    public Reserva getReserva() {
        return reserva;
    }



    public void setPrestado(boolean prestado) {
        this.prestado = prestado;
    }

    public void setReservado(boolean reservado) {
        this.reservado = reservado;
    }

    public void setRetraso(boolean retraso) {
        this.retraso = retraso;
    }

    public void setPrestamo(Prestamo prestamo) {
        this.prestamo = prestamo;
    }

    public void setReserva(Reserva reserva) {
        this.reserva = reserva;
    }
}

File: Python/gestorAplicacion/libreria/estadoEjemplar.py
from Python.gestorAplicacion.servicios.prestamo import Prestamo
from Python.gestorAplicacion.servicios.reserva import Reserva

class EstadoEjemplar():
    def __init__(self, prestado: bool, reservado: bool, retraso: bool, prestamo: Prestamo, reserva: Reserva):
        self._prestado = prestado
        self._reservado = reservado
        self._retraso = retraso
        self._prestamo = prestamo
        self._reserva = reserva

    def isPrestado(self) -> bool:
        return self._prestado

    def isReservado(self) -> bool:
        return self._reservado

    def isRetraso(self) -> bool:
        return self._retraso

    def getPrestamo(self) -> Prestamo:
        return self._prestamo

    def getReserva(self) -> Reserva:
        return self._reserva

    def setPrestado(self, prestado: bool):
        self._prestado = prestado

    def setReservado(self, reservado: bool):
        self._reservado = reservado

    def setRetraso(self, retraso: bool):
        self._retraso = retraso

    def setPrestamo(self, prestamo: Prestamo):
        self._prestamo = prestamo

    def setReserva(self, reserva: Reserva):
        self._reserva = reserva
